### PR TITLE
Map and Geocoding Service

### PR DESCRIPTION
**Pull request by JannisTriesToCode**
Just take a quick look pls.

### CODE REVIEW

File: services/maps/geocoding_service.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+from opencage.geocoder import OpenCageGeocode, InvalidInputError, RateLimitExceededError, UnknownError
+from services.Singleton import Singleton
+
+class GeocodingRemote(ABC):
+    @abstractmethod
+    def get_information_from_address(self, address):
```
Comment (Ovakefali13) [inline services/maps/geocoding_service.py]: please add the type of the variable like this `def get_information_from_address(self, address:str)`

File: services/maps/map_service.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+class MapRemote(ABC):
+    @abstractmethod
+    def get_route_information(self, start, dest, travel_mode=None):
+        pass
+
+@Singleton
+class MapJSONRemote(MapRemote):
+    # openroute service
+    API_TOKEN = '5b3ce3597851110001cf62488ef50484ef524d228826ecc7b35f5df1'
```
Comment (Ovakefali13) [inline services/maps/map_service.py]: Add token to the prefs ?

File: services/maps/geocoding_service.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        pass
+    @abstractmethod
+    def get_information_from_coords(self, coords):
+        pass
+
+@Singleton
+class GeocodingJSONRemote(GeocodingRemote):
+    # 2500 free transactions per day https://opencagedata.com
+    API_TOKEN = '2642206c773e4b06aedabd0a0e876a2f'
```
Comment (Ovakefali13) [inline services/maps/geocoding_service.py]: Add token to prefs ?

**Comment by JannisTriesToCode** [comment]
@Ovakefali13 I hope this suffices

**Comment by Ovakefali13** [comment]
Looks good

**Pull request merged**